File: qc-application/src/main/java/com/eager/questioncloud/application/business/payment/question/service/QuestionPaymentCouponService.java
package com.eager.questioncloud.application.business.payment.question.service;

import com.eager.questioncloud.application.business.payment.question.implement.QuestionPaymentCouponReader;
import com.eager.questioncloud.core.domain.payment.model.QuestionPaymentCoupon;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class QuestionPaymentCouponService {
    private final QuestionPaymentCouponReader questionPaymentCouponReader;

    public QuestionPaymentCoupon getQuestionPaymentCoupon(Long userCouponId, Long userId) {
        return questionPaymentCouponReader.getQuestionPaymentCoupon(userCouponId, userId);
    }
}

File: qc-api/qc-payment-api/src/main/kotlin/com/eager/questioncloud/payment/question/service/QuestionPaymentCouponService.kt
package com.eager.questioncloud.payment.question.service

import com.eager.questioncloud.payment.domain.QuestionPaymentCoupon
import com.eager.questioncloud.payment.question.implement.QuestionPaymentCouponReader
import org.springframework.stereotype.Service

@Service
class QuestionPaymentCouponService(
    private val questionPaymentCouponReader: QuestionPaymentCouponReader
) {
    fun getQuestionPaymentCoupon(userCouponId: Long?, userId: Long): QuestionPaymentCoupon? {
        return questionPaymentCouponReader.getQuestionPaymentCoupon(userCouponId, userId)
    }
}
